test/firtool/firtool.fir

```diff
@@ -277,3 +285,40 @@ circuit test_mod : %[[{"class": "circt.testNT", "data": "a"}]]
     input in : UInt<1>
     output out : UInt<1>
     out is invalid
+
+  intmodule Mux2:
+     input sel: UInt<1>
+     input high: UInt
+     input low: UInt
+     output out: UInt
+     intrinsic = circt_mux2cell
+
+  extmodule Val:
+     output v: UInt<1>
+  ; VERILOG-LABEL: Mux2Cell
+  ; Make sure that mux annotations are emitted properly.
+  ; Check that an assigned expression `~cond ? ... low` is not CSEd.
+  ; VERILOG:      /* synopsys infer_mux_override */
+  ; VERILOG-NEXT: assign [[OUT1:.+]] = ~cond ? /* cadence map_to_mux */ _ext_v : low;
+  ; VERILOG:      /* synopsys infer_mux_override */
+  ; VERILOG-NEXT: assign [[OUT2:.+]] = ~cond ? /* cadence map_to_mux */ _ext_v : low;
+  ; VERILOG: assign out_0 = [[OUT1]];
+  ; VERILOG: assign out_1 = [[OUT2]];
+
+  module Mux2Cell:
+    input cond: UInt<1>
+    input low: UInt<1>
+    output out: UInt<1>[2]
+    wire w: UInt<1>
+    inst mux2 of Mux2
+    inst ext of Val
+    w <= ext.v
+    mux2.sel <= xor(cond, UInt<1>(1))
+    mux2.high <= w
+    mux2.low <= low
+    out[0] <= mux2.out
+    inst mux2_1 of Mux2
+    mux2_1.sel <= xor(cond, UInt<1>(1))
+    mux2_1.high <= ext.v
+    mux2_1.low <= low
+    out[1] <= mux2_1.out
```

Comment:
This looks a bit complicated but this is an example that produces incorrect results if CSE prevention was not added.